texthero/representation.py

```diff
@@ -179,28 +205,41 @@ def count(
         max_df=max_df,
         binary=binary,
     )
-    s = pd.Series(tf.fit_transform(s).toarray().tolist(), index=s.index)
 
-    if return_feature_names:
-        return (s, tf.get_feature_names())
-    else:
-        return s
+    tf_vectors_csr = tf.fit_transform(s)
+    tf_vectors_coo = coo_matrix(tf_vectors_csr)
+
+    s_out = pd.Series.sparse.from_coo(tf_vectors_coo)
+
+    features_names = tf.get_feature_names()
+
+    # Map word index to word name
+    s_out.index = s_out.index.map(lambda x: (s.index[x[0]], features_names[x[1]]))
+
+    s_out.rename_axis(["document", "word"], inplace=True)
```

Comment:
I forgot. How axes are called if we don't rename them?

I'm starting seriously considering removing this line. Texthero purpose is to provide a tool and some guidelines on how to clean code, but also at the same time to stay enough flexible ... we might want to just mention in the _tutorials/getting-started_ that a good idea might be to add names to the indexes with `s.rename_axis(["document", "word"], inplace=True)` but don't force that. 

We might want to look at the Pandas codebase, but I doubt some of their functions add/rename axis ...

Expected behaviour: 

```
>>> hero.count(s)
    0         Sentence    1
              one         1
    1         Sentence    1
              two         1
    dtype: Sparse[int64, 0]
```

Reply:
They just don't have a name if we don't rename them. I agree it's unnecessary (and unexpected) for us to do this; showcasing it in a tutorial makes more sense. Will remove